packages\upgrade\static\src symlink doesn't work on Windows

<!--
IF YOU DON'T FILL OUT THE FOLLOWING INFORMATION WE MIGHT CLOSE YOUR ISSUE WITHOUT INVESTIGATING
-->

**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

`packages\upgrade\static\src`  is a symbolic link that isn't interpreted correctly on Windows. This results in error during compilation:

```
Compiling tools...
Creating .metadata.json files...
Error: Error at <...>/packages/upgrade/static/public_api.ts:15:44: Cannot find module './src/common/angular1'.
Error at <...>/packages/upgrade/static/public_api.ts:16:34: Cannot find module './src/common/downgrade_component'.
Error at <...>/packages/upgrade/static/public_api.ts:17:35: Cannot find module './src/common/downgrade_injectable'.
Error at <...>/packages/upgrade/static/public_api.ts:18:23: Cannot find module './src/common/version'.
Error at <...>/packages/upgrade/static/public_api.ts:19:32: Cannot find module './src/static/upgrade_component'.
Error at <...>/packages/upgrade/static/public_api.ts:20:29: Cannot find module './src/static/upgrade_module'.
    at UserError.Error (native)
    at new UserError (<...>\dist\tools\@angular\tsc-wrapped\src\tsc.js:21:28)
    at check (<...>\dist\tools\@angular\tsc-wrapped\src\tsc.js:87:15)
    at Tsc.typeCheck (<...>\dist\tools\@angular\tsc-wrapped\src\tsc.js:172:9)
    at <...>\dist\tools\@angular\tsc-wrapped\src\main.js:106:23
    at process._tickCallback (internal/process/next_tick.js:109:7)
    at Module.runMain (module.js:606:11)
    at run (bootstrap_node.js:390:7)
    at startup (bootstrap_node.js:150:9)
    at bootstrap_node.js:505:3
Compilation failed
```

**Expected behavior**

Symbolic link is supposed to work for all operation systems.

<!-- Describe what the behavior would be without the bug. -->

**Minimal reproduction of the problem with instructions**

In cloned repo:

npm i
bash ./scripts/windows/create-symlinks.sh 
bash ./test.sh node

**Please tell us about your environment:**
<!-- Operating system, IDE, package manager, HTTP server, ... -->
Windows 7
Node 6.10.3
NPM 3.10.10
git 2.9.0.windows.1

* **Angular version:** 4.2.0-beta.1
<!-- Check whether this is still an issue in the most recent Angular version -->

--- a/scripts/windows/create-symlinks.sh
+++ b/scripts/windows/create-symlinks.sh
@@ -2,6 +2,9 @@
 
 cd `dirname $0`
 
-DESTDIR=./../../packages/core/src/animation
-mv ${DESTDIR}/dsl.ts ${DESTDIR}/dsl.ts.old
+CORE_SRC_ANIMATION_DIR=./../../packages/core/src/animation
+UPGRADE_STATIC_DIR=./../../packages/upgrade/static
+mv ${CORE_SRC_ANIMATION_DIR}/dsl.ts ${CORE_SRC_ANIMATION_DIR}/dsl.ts.old
+mv ${UPGRADE_STATIC_DIR}/src ${UPGRADE_STATIC_DIR}/src.old
 cmd <<< "mklink \"..\\..\\packages\\core\\src\\animation\\dsl.ts\" \"..\\..\\..\\animations\\src\\animation_metadata.ts\""
+cmd <<< "mklink /d \"..\\..\\packages\\upgrade\\static\\src\" \"..\\src\""

--- a/scripts/windows/remove-symlinks.sh
+++ b/scripts/windows/remove-symlinks.sh
@@ -2,6 +2,9 @@
 
 cd `dirname $0`
 
-DESTDIR=./../../packages/core/src/animation
-rm ${DESTDIR}/dsl.ts
-mv ${DESTDIR}/dsl.ts.old ${DESTDIR}/dsl.ts
+CORE_SRC_ANIMATION_DIR=./../../packages/core/src/animation
+UPGRADE_STATIC_DIR=./../../packages/upgrade/static
+rm ${CORE_SRC_ANIMATION_DIR}/dsl.ts
+rm ${UPGRADE_STATIC_DIR}/src
+mv ${CORE_SRC_ANIMATION_DIR}/dsl.ts.old ${CORE_SRC_ANIMATION_DIR}/dsl.ts
+mv ${UPGRADE_STATIC_DIR}/src.old ${UPGRADE_STATIC_DIR}/src

